File: app/src/main/java/com/simcom/ecashier/model/room/CollectionDao.java
package com.simcom.ecashier.model.room;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;


@Dao
public interface CollectionDao {

    @Insert
    void insert(Collection collection);

    @Update
    void update(Collection collection);

    @Delete
    void delete(Collection collection);

    @Query("SELECT * FROM COLLECTION_TABLE where id = :id")
    LiveData<Collection> getCollection(int id);

    @Query("SELECT * FROM COLLECTION_TABLE ORDER BY time DESC")
    LiveData<List<Collection>> getAllCollections();

    @Query("SELECT * FROM COLLECTION_TABLE WHERE current = 1 LIMIT 1")
    LiveData<Collection> getCurrentCollection();


    @Query("SELECT * FROM COLLECTION_TABLE WHERE current = 1 LIMIT 1")
    Collection getCurrentCollectionClear();

    @Query("SELECT id AS collectionId, price, name, groupId, " +
            "(SELECT COUNT(*) FROM PERSON_TO_GROUP_TABLE AS ptg WHERE c.groupId = ptg.groupId) AS peopleCount, " +
            "(SELECT COUNT(*) FROM COLLECTION_LOG_TABLE AS cl WHERE cl.collectionId = c.id) AS paymentsCount " +
            "FROM COLLECTION_TABLE AS c WHERE c.current = 1 LIMIT 1")
    LiveData<CollectionInfo> getCurrentCollectionInfo();

}

File: app/src/main/java/com/simcom/ecashier/model/room/CollectionDao.kt
package com.simcom.ecashier.model.room

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface CollectionDao {
    @Insert
    public suspend fun insert(collection: Collection) : Long

    @Update
    public suspend fun update(collection: Collection)

    @Delete
    public suspend fun delete(collection: Collection)

    @Query("SELECT * FROM COLLECTION_TABLE where id = :id")
    public fun getCollection(id: Int): LiveData<Collection>

    @Query("SELECT * FROM COLLECTION_TABLE ORDER BY time DESC")
    public fun getAllCollections(): LiveData<List<Collection>>

    @Query("SELECT * FROM COLLECTION_TABLE WHERE isCurrent = 1 LIMIT 1")
    public fun getCurrentCollection(): LiveData<Collection>

    @Query("SELECT * FROM COLLECTION_TABLE WHERE isCurrent = 1 LIMIT 1")
    public fun getCurrentCollectionClear(): Collection

    @Query(
        "SELECT id AS collectionId, price, name, groupId, " +
                "(SELECT COUNT(*) FROM PERSON_TO_COLLECTION_TABLE AS ptc WHERE ptc.collectionId = c.id) AS peopleCount, " +
                "(SELECT COUNT(*) FROM PERSON_TO_COLLECTION_TABLE AS ptc WHERE ptc.collectionId = c.id AND ptc.hasPaid = 1) AS paymentsCount " +
                "FROM COLLECTION_TABLE AS c WHERE c.isCurrent = 1 LIMIT 1")
    public fun getCurrentCollectionInfo(): LiveData<CollectionInfo>

    @Query("UPDATE COLLECTION_TABLE SET isCurrent = 0")
    suspend fun makeNoCollectionCurrent()
}